core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/message/conversationlist/ConversationListViewModel.kt

```diff
@@ -1,13 +1,101 @@
 package com.ustadmobile.core.viewmodel.message.conversationlist
 
 import app.cash.paging.PagingSource
+import com.ustadmobile.core.MR
+import com.ustadmobile.core.impl.appstate.FabUiState
+import com.ustadmobile.core.impl.nav.UstadSavedStateHandle
 import com.ustadmobile.core.util.SortOrderOption
+import com.ustadmobile.core.view.ListViewMode
+import com.ustadmobile.core.view.UstadView
+import com.ustadmobile.core.viewmodel.UstadListViewModel
+import com.ustadmobile.core.viewmodel.message.messagelist.MessageListViewModel
+import com.ustadmobile.core.viewmodel.person.PersonViewModelConstants
 import com.ustadmobile.core.viewmodel.person.list.EmptyPagingSource
-import com.ustadmobile.lib.db.composites.MessageAndSenderPerson
+import com.ustadmobile.core.viewmodel.person.list.PersonListViewModel
+import com.ustadmobile.lib.db.composites.MessageAndOtherPerson
+import kotlinx.coroutines.flow.update
+import org.kodein.di.DI
 
 data class ConversationListUiState(
-    val conversations: () -> PagingSource<Int, MessageAndSenderPerson> = { EmptyPagingSource() },
-    val sortOptions: List<SortOrderOption> = emptyList() //Should be by name (ascending/descending), time (ascending/descending)
+    val conversations: () -> PagingSource<Int, MessageAndOtherPerson> = { EmptyPagingSource() },
+    val sortOptions: List<SortOrderOption> = emptyList(), //Should be by name (ascending/descending), time (ascending/descending)
+    val showAddItem: Boolean = false,
 )
 
+class ConversationListViewModel(
+    di: DI,
+    savedStateHandle: UstadSavedStateHandle,
+    destinationName: String = DEST_NAME,
+): UstadListViewModel<ConversationListUiState>(
+    di, savedStateHandle, ConversationListUiState(), destinationName
+) {
+
+    private val pagingSourceFactory: () -> PagingSource<Int, MessageAndOtherPerson> = {
+        activeRepo.messageDao.conversationsForUserAsPagingSource(
+           activeUserPersonUid
+        ).also {
+            lastPagingSource = it
+        }
+    }
+
+    private var lastPagingSource: PagingSource<Int, MessageAndOtherPerson>? = null
+
+    init {
+        _uiState.update { prev ->
+            prev.copy(
+                conversations = pagingSourceFactory
+            )
+        }
+
+        _appUiState.update { prev ->
+            prev.copy(
+                navigationVisible = true,
+                searchState = createSearchEnabledState(),
+                title = listTitle(MR.strings.messages, MR.strings.select_person),
+                fabState = FabUiState(
+                    visible = true,
+                    text = systemImpl.getString(MR.strings.message),
+                    icon = FabUiState.FabIcon.ADD,
+                    onClick = this@ConversationListViewModel::onClickAdd
+                )
+            )
+        }
+    }
+
+    override fun onUpdateSearchResult(searchText: String) {
+        TODO("Not yet implemented")
```

Comment:
This should be implemented as per other screens so that the user can search by the name of the person they are looking for. You can use the LIKE clause in the SQL query. 

e.g. 
```
WHERE searchQuery = '' OR ((Person.firstNames | ' ' | Person.lastName) LIKE searchQuery)
```